crypto/err/err.c

```diff
@@ -527,7 +531,8 @@ void ossl_err_string_int(unsigned long e, const char *func,
     }
 #endif
     if (rs == NULL) {
-        BIO_snprintf(rsbuf, sizeof(rsbuf), "reason(%lu)", r);
+        BIO_snprintf(rsbuf, sizeof(rsbuf), "reason(%lu)",
+                     r & ~(ERR_RFLAGS_MASK << ERR_RFLAGS_OFFSET));
```

Comment:
I would approve everything of this PR, apart for this particular change.  It's not strictly necessary with regards to missing string, which is the actual bug you're trying to fix.

I don't mean to say that this isn't a change worth thinking about, but it might not be suitable for 3.0...  so I would suggest moving this change to a different PR.

Reply:
Pleased to hear.
Moved that part to #17072.